Could not compile StructInit with json2.Any with OrBlock ?

CC version: Apple clang version 12.0.5 (clang-1205.0.22.11)

getwd: /Users/username/code/lang/vlang/vlang
vmodules: /Users/username/.vmodules
vroot: /Users/username/src/v
vexe: /Users/username/src/v/v
vexe mtime: 2021-09-21 02:52:24
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 a8905e2.b2ecca3

Git version: git version 2.33.0
Git vroot status: weekly.2021.37-51-gb2ecca39-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg study/issue/optional_sum_type_if_struct_init.v`
```v
import x.json2

struct SA {}
struct SB {}
struct SC {}
type Sum = SA | SB | SC
struct App {
	s map[string]Sum
	t bool
}
fn main() {
	mut m := map[string]json2.Any{}
	app := App {
		t: m['t'] or { 0 }.bool()
		s: if a := m['a'] {
			println('a => $a')
			b := return_optional(a) ? // Fails only if the expr in this line has or_block (or_block kind (.propagation or .block) doesn't matter)
			b
		} else {
			map[string]Sum{}
		}
	}
	println('app => $app')
}

fn return_optional(input json2.Any) ?map[string]Sum {
	return map[string]Sum{}
}
```

**What did you expect to see?**

successfully built!


**What did you see instead?**
```
==================
/tmp/v/optional_sum_type_if_struct_init.13928218203421388152.tmp.c:11814:81: error: use of undeclared identifier '_t5'
        main__App app = (main__App){.s =  _t1,.t = x__json2__Any_bool(*(x__json2__Any*)_t5.data),};
                                                                                       ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4799,6 +4799,7 @@ fn (mut g Gen) if_expr(node ast.IfExpr) {
 		}
 	}
 	g.writeln('}')
+	g.stmt_path_pos << g.out.len
 	if needs_tmp_var {
 		if g.infix_left_var_name.len > 0 {
 			g.indent--

--- a/vlib/v/tests/init_multiple_branches_test.v
+++ b/vlib/v/tests/init_multiple_branches_test.v
@@ -1,0 +1,29 @@
+// fixes https://github.com/vlang/v/issues/11564, test is copied from the code by https://github.com/ken0x0a
+import x.json2
+
+struct SA {}
+struct SB {}
+struct SC {}
+type Sum = SA | SB | SC
+struct App {
+	s map[string]Sum
+	t bool
+}
+fn test_init_multiple_branches() ? {
+	mut m := map[string]json2.Any{}
+	app := App {
+		t: m['t'] or { 0 }.bool()
+		s: if a := m['a'] {
+			println('a => $a')
+			b := return_optional(a) ? // Fails only if the expr in this line has or_block (or_block kind (.propagation or .block) doesn't matter)
+			b
+		} else {
+			map[string]Sum{}
+		}
+	}
+	println('app => $app')
+}
+
+fn return_optional(input json2.Any) ?map[string]Sum {
+	return map[string]Sum{}
+}

--- a/vlib/v/tests/init_multiple_branches_test.v
+++ b/vlib/v/tests/init_multiple_branches_test.v
@@ -1,17 +1,22 @@
-// fixes https://github.com/vlang/v/issues/11564, test is copied from the code by https://github.com/ken0x0a
+// fixes https://github.com/vlang/v/issues/11564, test is copied from the code by https://github.com/ken0x0a and formatted
 import x.json2
 
 struct SA {}
+
 struct SB {}
+
 struct SC {}
+
 type Sum = SA | SB | SC
+
 struct App {
 	s map[string]Sum
 	t bool
 }
+
 fn test_init_multiple_branches() ? {
 	mut m := map[string]json2.Any{}
-	app := App {
+	app := App{
 		t: m['t'] or { 0 }.bool()
 		s: if a := m['a'] {
 			println('a => $a')

